packages/sensors_plus/tizen/src/sensors_plus_plugin.cc

```diff
@@ -3,221 +3,89 @@
-              if (gyroscope_listener_) {
-                gyroscope_listener_->Cancel();
-                gyroscope_listener_ = nullptr;
-              }
-              return nullptr;
-            });
-    gyroscope_channel_->SetStreamHandler(std::move(gyroscope_channel_handler));
-
-    user_accel_channel_ =
-        std::make_unique<flutter::EventChannel<flutter::EncodableValue>>(
-            registrar->messenger(),
-            "dev.fluttercommunity.plus/sensors/user_accel",
-            &flutter::StandardMethodCodec::GetInstance());
-    auto user_accel_handler =
-        std::make_unique<flutter::StreamHandlerFunctions<>>(
-            [this](const flutter::EncodableValue *arguments,
-                   std::unique_ptr<flutter::EventSink<>> &&events)
-                -> std::unique_ptr<flutter::StreamHandlerError<>> {
-              LOG_DEBUG("OnListen");
-              user_accel_listener_ =
-                  std::make_unique<Listener>(std::move(events));
-              if (user_accel_listener_->Init(SENSOR_LINEAR_ACCELERATION)) {
-                user_accel_listener_->Listen();
-              }
-              return nullptr;
-            },
-            [this](const flutter::EncodableValue *arguments)
-                -> std::unique_ptr<flutter::StreamHandlerError<>> {
-              LOG_DEBUG("OnCancel");
-              if (user_accel_listener_) {
-                user_accel_listener_->Cancel();
-                user_accel_listener_ = nullptr;
-              }
-              return nullptr;
-            });
-    user_accel_channel_->SetStreamHandler(std::move(user_accel_handler));
+        std::make_unique<DeviceSensorStreamHandler>(
+            SensorType::kAccelerometer));
+
+    gyroscope_channel_ = std::make_unique<FlEventChannel>(
+        registrar->messenger(), "dev.fluttercommunity.plus/sensors/gyroscope",
+        &flutter::StandardMethodCodec::GetInstance());
+    gyroscope_channel_->SetStreamHandler(
+        std::make_unique<DeviceSensorStreamHandler>(SensorType::kGyroscope));
+
+    user_accel_channel_ = std::make_unique<FlEventChannel>(
+        registrar->messenger(), "dev.fluttercommunity.plus/sensors/user_accel",
+        &flutter::StandardMethodCodec::GetInstance());
+    user_accel_channel_->SetStreamHandler(
+        std::make_unique<DeviceSensorStreamHandler>(SensorType::kUserAccel));
   }
 
-  std::unique_ptr<flutter::EventChannel<flutter::EncodableValue>>
-      accelerometer_channel_;
-  std::unique_ptr<flutter::EventChannel<flutter::EncodableValue>>
-      gyroscope_channel_;
-  std::unique_ptr<flutter::EventChannel<flutter::EncodableValue>>
-      user_accel_channel_;
-  std::unique_ptr<Listener> accelerometer_listener_;
-  std::unique_ptr<Listener> gyroscope_listener_;
-  std::unique_ptr<Listener> user_accel_listener_;
+  std::unique_ptr<FlEventChannel> accelerometer_channel_;
+  std::unique_ptr<FlEventChannel> gyroscope_channel_;
+  std::unique_ptr<FlEventChannel> user_accel_channel_;
```

Comment:
As far as I can remember, this has to do with number #131 and #134.
But a lot of time has passed since then, and I haven't investigated whether this patch is still necessary.

Reply:
Thank you for reminding me. The patch still seems relevant. I'll file a PR that reverts changes in https://github.com/flutter-tizen/plugins/pull/351 and https://github.com/flutter-tizen/plugins/pull/352.

Reply:
Fixed in https://github.com/flutter-tizen/plugins/pull/361 and https://github.com/flutter-tizen/plugins/pull/362.